File: backend/agents/market/app/schemas/market.py
from pydantic import BaseModel, Field
from typing import List, Dict, Any, Optional
from datetime import datetime

class MarketAnalysisResponse(BaseModel):
    agent: str = "Market Intelligence"
    marketRiskScore: int = Field(..., alias="marketRiskScore")
    competitorThreat: str = Field(..., alias="competitorThreat")
    marketOpportunity: str = Field(..., alias="marketOpportunity")
    demandForecast: str = Field(..., alias="demandForecast")
    confidence: int
    keyFindings: List[str] = Field(..., alias="keyFindings")
    recommendations: List[str] = Field(..., alias="recommendations")

    class Config:
        populate_by_name = True
        json_schema_extra = {
            "example": {
                "agent": "Market Intelligence",
                "marketRiskScore": 82,
                "competitorThreat": "High",
                "marketOpportunity": "Medium",
                "demandForecast": "Demand expected to increase by 12%",
                "confidence": 95,
                "keyFindings": ["Competitor Alpha launched a premium tier feature", "Economic inflation dropping slightly"],
                "recommendations": ["Optimize standard tier pricing", "Target edge computing niche market"]
            }
        }

class CompetitorDataSchema(BaseModel):
    competitor_id: int
    name: str
    product_name: str
    feature_set: str
    market_share: float
    pricing_tier: str
    sentiment_score: float
    last_launch_date: str
    status: str

class NewsItemSchema(BaseModel):
    date: str
    title: str
    summary: str
    source: str
    sentiment: str
    impact_score: int

class PricingCompareSchema(BaseModel):
    competitor_name: str
    plan_name: str
    price_monthly: float
    features_included: str

class TrendItemSchema(BaseModel):
    quarter: str
    trend_name: str
    adoption_rate: float
    growth_rate: float
    sentiment: str
    primary_driver: str

class DemandPointSchema(BaseModel):
    month: str
    sales_units: int
    nps: int
    lead_conversion_rate: float

class DemandForecastSchema(BaseModel):
    historical: List[DemandPointSchema]
    forecast: List[DemandPointSchema]
    message: str

class DashboardDataResponse(BaseModel):
    marketRiskScore: int
    competitorThreat: str
    marketOpportunity: str
    demandForecastSummary: str
    confidence: int
    competitors: List[CompetitorDataSchema]
    news: List[NewsItemSchema]
    pricing: List[PricingCompareSchema]
    trends: List[TrendItemSchema]
    demandForecast: DemandForecastSchema
    recommendations: List[str]

class RiskDetailsResponse(BaseModel):
    overallScore: int
    competitorThreatScore: int
    economicRiskScore: int
    demandRiskScore: int
    riskFactors: List[Dict[str, Any]]

class OpportunityDetailsResponse(BaseModel):
    overallOpportunityScore: int
    opportunities: List[Dict[str, Any]]

class CrisisAnalysisRequest(BaseModel):
    company_name: str
    industry: str
    crisis_type: str
    crisis_description: str
    current_market_trend: str
    competitor_information: str
    customer_demand: str
    location: str

class CrisisAnalysisResponse(BaseModel):
    market_summary: str
    market_impact: str
    competitor_analysis: str
    customer_demand_prediction: str
    business_opportunities: List[str]
    market_risk_score: str
    recommendations: List[str]

File: market-intelligence-agent/frontend/src/types/market.ts
export interface CompetitorData {
  competitor_id: number;
  name: string;
  product_name: string;
  feature_set: string;
  market_share: number;
  pricing_tier: string;
  sentiment_score: number;
  last_launch_date: string;
  status: string;
}

export interface NewsItem {
  date: string;
  title: string;
  summary: string;
  source: string;
  sentiment: string;
  impact_score: number;
}

export interface PricingCompare {
  competitor_name: string;
  plan_name: string;
  price_monthly: number;
  features_included: string;
}

export interface TrendItem {
  quarter: string;
  trend_name: string;
  adoption_rate: number;
  growth_rate: number;
  sentiment: string;
  primary_driver: string;
}

export interface DemandPoint {
  month: string;
  sales_units: number;
  nps: number;
  lead_conversion_rate: number;
}

export interface DemandForecast {
  historical: DemandPoint[];
  forecast: DemandPoint[];
  message: string;
}

export interface DashboardData {
  marketRiskScore: number;
  competitorThreat: string;
  marketOpportunity: string;
  demandForecastSummary: string;
  confidence: number;
  competitors: CompetitorData[];
  news: NewsItem[];
  pricing: PricingCompare[];
  trends: TrendItem[];
  demandForecast: DemandForecast;
  recommendations: string[];
}

export interface RiskFactor {
  factor: string;
  description: string;
  severity: string;
}

export interface RiskDetails {
  overallScore: number;
  competitorThreatScore: number;
  economicRiskScore: number;
  demandRiskScore: number;
  riskFactors: RiskFactor[];
}

export interface OpportunityItem {
  opportunity: string;
  description: string;
  impact: string;
}

export interface OpportunityDetails {
  overallOpportunityScore: number;
  opportunities: OpportunityItem[];
}

export interface ReportDetails {
  id: number;
  timestamp: string;
  agent: string;
  marketRiskScore: number;
  competitorThreat: string;
  marketOpportunity: string;
  demandForecast: string;
  confidence: number;
  keyFindings: string[];
  recommendations: string[];
}

export interface CrisisAnalysisRequest {
  company_name: string;
  industry: string;
  crisis_type: string;
  crisis_description: string;
  current_market_trend: string;
  competitor_information: string;
  customer_demand: string;
  location: string;
}

export interface CrisisAnalysisResponse {
  market_summary: string;
  market_impact: string;
  competitor_analysis: string;
  customer_demand_prediction: string;
  business_opportunities: string[];
  market_risk_score: string;
  recommendations: string[];
}

